homeassistant/components/fritz/config_flow.py

```diff
@@ -0,0 +1,248 @@
+        self._host = ssdp_location.hostname
+        self._port = ssdp_location.port
+        self._name = discovery_info.get(ATTR_UPNP_FRIENDLY_NAME)
+        self.context[CONF_HOST] = self._host
+
+        uuid = discovery_info.get(ATTR_UPNP_UDN)
+        if uuid:
+            if uuid.startswith("uuid:"):
+                uuid = uuid[5:]
+            await self.async_set_unique_id(uuid)
+            self._abort_if_unique_id_configured({CONF_HOST: self._host})
+
+        for progress in self._async_in_progress():
+            if progress.get("context", {}).get(CONF_HOST) == self._host:
+                return self.async_abort(reason="already_in_progress")
+
+        for entry in self._async_current_entries(include_ignore=False):
+            if entry.data[CONF_HOST] == self._host:
+                if uuid and not entry.unique_id:
+                    self.hass.config_entries.async_update_entry(entry, unique_id=uuid)
+                return self.async_abort(reason="already_configured")
+
+        self.context["title_placeholders"] = {
+            "name": self._name.replace("FRITZ!Box ", "")
+        }
+        return await self.async_step_confirm()
+
+    async def async_step_confirm(self, user_input=None):
+        """Handle user-confirmation of discovered node."""
+
+        if user_input is None:
+            return self._show_setup_form_confirm()
+
+        errors = {}
+
+        self._username = user_input[CONF_USERNAME]
+        self._password = user_input[CONF_PASSWORD]
+
+        error = await self.fritz_tools_init()
+
+        if error:
+            errors["base"] = error
+            return self._show_setup_form_confirm(errors)
+
+        return self._async_create_entry()
+
+    def _show_setup_form_init(self, errors=None):
+        """Show the setup form to the user."""
+        return self.async_show_form(
+            step_id="start_config",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(CONF_HOST, default=DEFAULT_HOST): str,
+                    vol.Optional(CONF_PORT, default=DEFAULT_PORT): vol.Coerce(int),
+                    vol.Required(CONF_USERNAME): str,
+                    vol.Required(CONF_PASSWORD): str,
+                }
+            ),
+            errors=errors or {},
+        )
+
+    def _show_setup_form_confirm(self, errors=None):
+        """Show the setup form to the user."""
+        return self.async_show_form(
+            step_id="confirm",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_USERNAME): str,
+                    vol.Required(CONF_PASSWORD): str,
+                }
+            ),
+            description_placeholders={"name": self._name},
+            errors=errors or {},
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        return await self.async_step_start_config()
```

Comment:
Can we rename `async_step_start_config` into `async_step_user` and delete what is in `async_step_user` now?